README.md

```diff
@@ -59,13 +59,13 @@ nvm is a version manager for [node.js](https://nodejs.org/en/), designed to be i
 
 To **install** or **update** nvm, you should run the [install script][2]. To do that, you may either download and run the script manually, or use the following cURL or Wget command:
 ```sh
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | $SHELL
```

Comment:
`install.sh` is written for bash only, so it won't necessarily work in every shell :-/

Reply:
@ljharb I totally agree. This is explicitly stated in the shebang like. Changing it to anything else risks breaking backward compatibility.

Reply:
I dont know if it could help, I use this to know what is the shell, not sure it works well for all the cases but I use it with different collegues using macOS, Linux, Windows WSL, MSYS or Cygwin.

`system_get_current_shell_path() {
  [[ -e "/proc/$$/exe" ]] && readlink "/proc/$$/exe" && return 0
  ps -p $$ | tr '[:space:]' '\n' | grep 'sh' | head -1 && return 0
  which "$SHELL" && return 0
  echo "$SHELL"
}

system_get_current_shell() {
  echo "$(basename "$(system_get_current_shell_path)" .exe)"
}
`

Reply:
> `install.sh` is written for bash only, so it won't necessarily work in every shell :-/

Ah got it. It worked well for me with my default being zsh, but I didn't test with other shells. I thought the install script would work with all as there were no notes in the readme about which ones to use. Which shells won't be able to run the installer?

The curl command is confusing users as the correct profile isn't updated. The problem seems to be exacerbated with the switch on macOS to zsh.

Reply:
dash, sh, ksh.

The solution is to improve `install.sh`, not to run it in "not bash".

Reply:
Got it. Shall I update the pr to only make the informational changes such as the ones you pointed out as being helpful? I can also add some notes about install.sh only running in dash, sh and ksh for now, and example curl commands for zsh and others that set the PROFILE env var so that the proper file is updated.

Reply:
install.sh is only meant to ever run in bash :-) keeping the informational part is great tho.